src/JsonArrayPrettyType.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Mapado\PrettyTypes;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\JsonArrayType;
+
+/**
+ * Array Type which can be used to generate pretty printed json arrays.
+ *
+ * @author Thomas di Luccio <thomas.diluccio@mapado.com>
+ */
+class JsonArrayPrettyType extends JsonArrayType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform)
+    {
+        if (null === $value) {
+            return null;
+        }
```

Comment:
By adding this, you change the comportment of JsonArray as it does not return a null value in the base class.

It does not throw ConversionException neither.

Reply:
humm. We still got `JsonArrayType extends JsonType` in 2.6.3. `JsonArrayType` doesn't have an explicit `convertToDatabaseValue` method. It's therefore more of a kid of dirty copy/paste of the dbal code. But not a the 2.5 version, indeed.